### PR TITLE
Mm update

### PR DESCRIPTION
**Pull request by Aragar199**
* Update window.web3 to window.ethereum
* Update window.ethereum.request from deprecated sendAsync
* update provider detection to use @metamask/detect-provider and set current account logic
* update balance - get current balance of selected account
                              - display in ether instead of Wei for visibility
                              - remove address argument to reflect metamask current account 
* update personal_sign - fix callback function
                                         - change personal signed result process to verify current account is signed verifier using eth-sig-util
* update sendTransaction - fix callback function
                                             - change input value to ether from wei
* update signedTypeData - fix callback function

### CODE REVIEW

File: src/nodes/Components/MetaMask.js [hunk 1]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
-    this.setOutputData(0,this.accounts[0])
-  }
-
+  this.connectWeb3()
 
   this.setOutputData(1,{
     name:"balance",
-    args:[{name:"address",type:"string"}],
-    function:async (args)=>{
+    function:async ()=>{
```
Comment (Aragar199) [inline src/nodes/Components/MetaMask.js:52]: Changed this to no longer get inputs from user and only use the current selected MM address.

File: src/nodes/Components/MetaMask.js [hunk 2]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
       try{
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
-        let balance = await currentWeb3.eth.getBalance(args.address)
-        return balance
+        console.log(this.accounts[0])
+        this.Web3 = new Web3(window.ethereum)
+        let balance = await this.Web3.eth.getBalance(this.accounts[0])
+        let eth_balance = this.Web3.utils.fromWei(balance)
```
Comment (Aragar199) [inline src/nodes/Components/MetaMask.js:58]: Change to display in Eth instead of Wei so easier to read.

File: src/nodes/Components/MetaMask.js [hunk 3]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
     }
   })
+
+
   this.setOutputData(2,{
     name:"sign",
     args:[{name:"message",type:"string"}],
     function:async (args)=>{
-      return new Promise((resolve, reject) => {
+      await new Promise ((resolve, reject) => {
```
Comment (Aragar199) [inline src/nodes/Components/MetaMask.js:71]: Function.js was closing the loop without firing callback, changed this to fix.

File: src/nodes/Components/MetaMask.js [hunk 4]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
 
-        if(typeof args.to != "undefined" && args.to!=null ){
+        if(typeof args.to !== "undefined" && args.to!=null ){
           transactionParameters.to = args.to
         }
 
-        if(typeof args.value != "undefined" && args.value!=null ){
-          transactionParameters.value = ""+currentWeb3.utils.toHex(args.value)
+        if(typeof args.value !== "undefined" && args.value!=null ){
+          let value = currentWeb3.utils.toWei(args.value.toString(), 'ether')
```
Comment (Aragar199) [inline src/nodes/Components/MetaMask.js:127]: Changed this input value to be in ether so easier to use than typing ether total in Wei.

File: src/nodes/Components/MetaMask.js [hunk 5]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
     name:"sign",
     args:[{name:"message",type:"string"}],
     function:async (args)=>{
-      return new Promise((resolve, reject) => {
+      await new Promise ((resolve, reject) => {
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
-        window.ethereum.sendAsync({
+        const from = this.accounts[0]
+        window.ethereum.request({
```
Comment (Aragar199) [inline src/nodes/Components/MetaMask.js:74]: sendAsync was deprecated.

**Comment by grahamtallen** [comment]
🎸 🎸 🎊

File: src/nodes/Components/MetaMask.js [hunk 6]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
     name:"balance",
-    args:[{name:"address",type:"string"}],
-    function:async (args)=>{
+    function:async ()=>{
       try{
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
-        let balance = await currentWeb3.eth.getBalance(args.address)
-        return balance
+        console.log(this.accounts[0])
```
Comment (grahamtallen) [inline src/nodes/Components/MetaMask.js:55]: @Aragar199 can you rm log

File: src/nodes/Components/MetaMask.js [hunk 7]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
+          console.dir({ recovered })
+
+          if (recovered === from ) {
+            console.log('SigUtil Successfully verified signer as ' + from)
+            resolve(window.alert('SigUtil Successfully verified signer as ' + from))
+          } else {
+            console.dir(recovered)
+            console.log('SigUtil Failed to verify signer when comparing ' + recovered.result + ' to ' + from)
+            reject(console.log('Failed, comparing %s to %s', recovered, from))
+          }
```
Comment (grahamtallen) [inline src/nodes/Components/MetaMask.js:96]: ah, I see a lot of these logs already exist. We can remove later as a part of linting. They are also very useful for developers, maybe we could start by setting them all to console.debug unless needed

File: src/nodes/Components/MetaMask.js [hunk 8]
```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
...
           //to: args.to, // Required except during contract publications.
-          from: window.ethereum.selectedAddress, // must match user's active address.
+          from: this.accounts[0], // must match user's active address.
           //value: ""+args.value, // Only required to send ether to the recipient from the initiating external account.
           //data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057', // Optional, but used for defining smart contract creation and interaction.
           //chainId: 3 // Used to prevent transaction reuse across blockchains. Auto-filled by MetaMask.
         }
 
-        if(typeof args.to != "undefined" && args.to!=null ){
+        if(typeof args.to !== "undefined" && args.to!=null ){
```
Comment (grahamtallen) [inline src/nodes/Components/MetaMask.js:122]: This guy still lints

**Pull request merged**